### PR TITLE
Fix Blocky UI behavior while scanning

### PR DESCRIPTION
**Pull request by iankchristie**
onUrlDeviceDiscoveryUpdate being called rapidly during scanning.
Because of inefficiencies in the process of adding it to the UI
this caused blocky behavior. This fix optimizes when we call the
UIThread to empty the GroupIdQueue.

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 1]
```diff
@@ -296,32 +296,33 @@ public void onListItemClick(ListView l, View v, int position, long id) {
 
   @Override
   public void onUrlDeviceDiscoveryUpdate() {
+    Log.v(TAG, "onUrlDeviceDiscoveryUpdate");
+    if (SwipeDismissListViewTouchListener.isLocked()) {
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: I think this check needs to happen on the UI thread, and it's already handled by emptyGroupIdQueue, so just remove it.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java [hunk 2]
```diff
@@ -296,32 +296,33 @@ public void onListItemClick(ListView l, View v, int position, long id) {
...
+      Log.d(TAG, "groupid to add " + groupId);
+      if (mNearbyDeviceAdapter.containsGroupId(groupId)) {
+        mNearbyDeviceAdapter.updateItem(pwPair);
+      } else if (!mGroupIdQueue.contains(groupId)
+          && !Utils.isBlocked(pwPair)) {
+        mGroupIdQueue.add(groupId);
+      }
+    }
+    // If queue is empty or the second scan is not complete, do not
+    // try to empty queue. Causes blocky UI behavior
```
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java]: Remove the comment about blocky UI behavior...that's not the core reason to check, and it shouldn't inherently cause janky behavior.

**Comment by iankchristie** [comment]
Is this fix ready to go?

**Comment by cco3** [comment]
I think so.  Please squash the changes together first.

**Comment by iankchristie** [comment]
@cco3 squashed the changes. Credit: Doug & Matt

**Pull request merged**